examples/solvers/chambolle_pock_deconvolve.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Deconvolution using Chambolle-Pock solver."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+
+standard_library.install_aliases()
+from builtins import super
+
+# External
+import numpy as np
+import scipy
+import scipy.ndimage
+import matplotlib.pyplot as plt
+
+# Internal
+import odl
+
+
+# TODO: Use BroadCastOperator instead of ProductSpaceOperator
+
+class Convolution(odl.Operator):
+    def __init__(self, space, kernel, adjkernel):
+        self.kernel = kernel
+        self.adjkernel = adjkernel
+        super().__init__(space, space, linear=True)
+
+    def _call(self, rhs, out):
+        scipy.ndimage.convolve(rhs, self.kernel, output=out.asarray(),
+                               mode='wrap')
+        return out
+
+    @property
+    def adjoint(self):
+        return Convolution(self.domain, self.adjkernel, self.kernel)
+
+
+def kernel(x):
+    mean = [0.1, 0.5]
+    std = [0.1, 0.1]
+    k = np.exp(-(((x[0] - mean[0]) / std[0]) ** 2 +
+                 ((x[1] - mean[1]) / std[1]) ** 2))
+    k = np.around(k, 8)
```

Comment:
Forgot to remove that. That was left when Chong and I were testing Chambolle-Pock and looking at the kernel.